packages/api-plugin-fulfillment-method-pickup-store/src/preStartup.js

```diff
@@ -0,0 +1,27 @@
+import { MethodStoreData } from "./simpleSchemas.js";
+/**
+ * @summary Called on preStartup
+ * @param {Object} context Startup context
+ * @returns {undefined}
+ */
+export default async function fulfillmentMethodPickupStorePreStartup(context) {
+  const { simpleSchemas: { ShippingMethod, SelectedFulfillmentOption } } = context;
+
+  ShippingMethod.extend({
+    methodAdditionalData: {
+      type: ShippingMethod.getDefinition(
+        "methodAdditionalData",
+        ["type"]
+      ).type[0].type.extend(MethodStoreData)
+    }
+  });
+
+  SelectedFulfillmentOption.extend({
+    methodAdditionalData: {
+      type: SelectedFulfillmentOption.getDefinition(
+        "methodAdditionalData",
+        ["type"]
+      ).type[0].type.extend(MethodStoreData)
+    }
+  });
```

Comment:
Slightly better architecture might be to pass these in the `registerPlugin` object:

```js
{
  fulfillmentProviders: [
    {
      type: "pickup",
      name: "store",
      methodAdditionalDataSchema: MethodStoreData
    }
  ]
}
```

Then use that information in the cart plugin or wherever `ShippingMethod`, etc. schemas are defined.

In general I prefer the pattern of providing information from more specific plugins to less specific plugins, and let the less specific plugins extend their own schemas based on that information. Otherwise the less specific plugin may be confused about how its schemas have been altered.

Reply:
This will be done after the merge of this branch to feat/fulfillment-type
issue for tracking [here](https://app.zenhub.com/workspaces/moc-main-board-62fd8e881686130027eafc31/issues/gh/reactioncommerce/reaction/6822)